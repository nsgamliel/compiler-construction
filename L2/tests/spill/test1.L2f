(:myF
0

  %myVar <- 0
  %myVar++
  %myVar += 3
  %myVar += 5
%variable6GXZOSEK7I <- %myVar

rdi <- %variable6GXZOSEK7I
  call :myF2 0

  return
)

%myVar

%S
