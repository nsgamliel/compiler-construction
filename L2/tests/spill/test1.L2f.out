(:myF
	0 1
	%S0 <- 0
	mem rsp 0 <- %S0
	%S1 <- mem rsp 0
	%S1++
	mem rsp 0 <- %S1
	%S2 <- mem rsp 0
	%S2 += 3
	mem rsp 0 <- %S2
	%S3 <- mem rsp 0
	%S3 += 5
	mem rsp 0 <- %S3
	%S4 <- mem rsp 0
	%variable6GXZOSEK7I <- %S4
	rdi <- %variable6GXZOSEK7I
	call :myF2 0
	return
)
